lib/install/deps.js

```diff
@@ -74,7 +74,7 @@ function doesChildVersionMatch (child, requested, requestor) {
     return false
   }
   try {
-    return semver.satisfies(child.package.version, requested.fetchSpec)
+    return semver.satisfies(child.package.version, requested.fetchSpec, true)
```

Comment:
@destroyerofbuilds This is [Postel's law](https://en.wikipedia.org/wiki/Robustness_principle) in action, and it's a pretty good principle. It is much better for npm to keep _working_ even when users get "creative". And, as @BridgeAR point out, the "loose" option is documented and has been there for a while -- and npm has been using it for a long time. This patch isn't a new thing. It's fixing a regression because for some reason we _stopped_ using loose here.